app/models/manageiq/providers/redhat/infra_manager/admin_ui.rb

```diff
@@ -0,0 +1,96 @@
+module ManageIQ::Providers::Redhat::InfraManager::AdminUI
+  extend ActiveSupport::Concern
+
+  require 'openssl'
+  require 'ovirtsdk4'
+  require 'uri'
+  require 'json'
+
+  def queue_generate_admin_ui_url
+    task_opts = {
+      :action => "Generate oVirt Admin UI URL for EMS"
+    }
+
+    queue_opts = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "generate_admin_ui_url",
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => "ems_operations",
+      :zone        => my_zone,
+      :args        => []
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def generate_admin_ui_url
+    $rhevm_log.info("Generating oVirt Admin UI URL for EMS with identifier '#{id}'.")
+
+    # Get the credentials of the provider:
+    user = authentication_userid(:default)
+    password = authentication_password(:default)
+
+    # Create the HTTP client:
+    insecure = default_endpoint.verify_ssl == OpenSSL::SSL::VERIFY_NONE,
+    ca_certs = default_endpoint.certificate_authority
+    ca_certs = nil if ca_certs.blank?
+    ca_certs = [ca_certs] if ca_certs
+    client = OvirtSDK4::HttpClient.new(
+      :insecure => insecure,
+      :ca_certs => ca_certs,
+      :log      => $rhevm_log
+    )
+
+    # Request new SSO token using the provider credentials:
+    request = OvirtSDK4::HttpRequest.new
+    request.method = :GET
```

Comment:
If we can use `:POST` instead of `:GET` and put the parameters in the request body, then there will be no concern about having the user name and password in the logs. That may require changes in the SSO service, not sure if it supports both `POST` and `GET` currently.

Reply:
I'll test this with oVirt Engine master.